Peer dependency failure with beta storybook

**Describe the bug**

I am using storybook @ 6.3.0-beta.2, and when trying to install @storybook/testing-react with npm 7 I get the following:

```
pm ERR! Found: @storybook/addons@6.3.0-beta.2
npm ERR! node_modules/@storybook/addons
npm ERR!   dev @storybook/addons@"^6.3.0-beta.2" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @storybook/addons@"^6.0.0" from @storybook/testing-react@0.0.16
npm ERR! node_modules/@storybook/testing-react
npm ERR!   dev @storybook/testing-react@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```


**To Reproduce**
Steps to reproduce the behavior:
Install storybook beta version, use npm 7, try to install @storybook/testing-react.

**Expected behavior**
It should install successfully.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
I propose setting the peer dependency ranges as loosely as possible, perhaps:

```
  "peerDependencies": {
    "react": ">=16.8.0",
    "@storybook/react": ">=6.0.0",
    "@storybook/addons": ">=6.0.0",
    "@storybook/client-api": ">=6.0.0"
  },
```

--- a/package.json
+++ b/package.json
@@ -70,10 +70,10 @@
     "typescript": "^4.2.3"
   },
   "peerDependencies": {
-    "react": ">=16.8.0",
-    "@storybook/react": ">=6.0.0",
-    "@storybook/addons": ">=6.0.0",
-    "@storybook/client-api": ">=6.0.0"
+    "react": "^16.8.0 || ^17.0.0",
+    "@storybook/react": "^6 || >=6.0.0-0 || >=6.1.0-0 || >=6.2.0-0 || >=6.3.0-0 ",
+    "@storybook/addons": "^6 || >=6.0.0-0 || >=6.1.0-0 || >=6.2.0-0 || >=6.3.0-0 ",
+    "@storybook/client-api": "^6 || >=6.0.0-0 || >=6.1.0-0 || >=6.2.0-0 || >=6.3.0-0 "
   },
   "resolutions": {
     "**/typescript": "^4.2.3"

--- a/package.json
+++ b/package.json
@@ -70,10 +70,10 @@
     "typescript": "^4.2.3"
   },
   "peerDependencies": {
-    "react": "^16.8.0 || ^17.0.0",
-    "@storybook/react": "^6.0.0",
-    "@storybook/addons": "^6.0.0",
-    "@storybook/client-api": "^6.0.0"
+    "react": ">=16.8.0",
+    "@storybook/react": ">=6.0.0",
+    "@storybook/addons": ">=6.0.0",
+    "@storybook/client-api": ">=6.0.0"
   },
   "resolutions": {
     "**/typescript": "^4.2.3"

